tests/component/test_stream_writers_ao.py

```diff
@@ -0,0 +1,393 @@
+import ctypes
+
+import numpy
+import pytest
+
+import nidaqmx
+from nidaqmx.stream_writers import (
+    AnalogMultiChannelWriter,
+    AnalogSingleChannelWriter,
+    AnalogUnscaledWriter,
+)
+
+
+def _get_expected_voltage_for_chan(chan_index: int) -> float:
+    return float(chan_index + 1)
+
+
+def _volts_to_codes(volts: float, max_code: int = 32767, max_voltage: float = 10.1) -> int:
+    return int(volts * max_code / max_voltage)
+
+
+VOLTAGE_EPSILON = 1e-3
+# NOTE: You can't scale from volts to codes correctly without knowing the internal calibration
+# constants. The internal reference has a healthy amount of overrange to ensure we can calibrate to
+# device specifications. I've used 10.1 volts above to approximate that, but 100mv of accuracy is
+# also fine since the expected output of each channel value will be 1 volt apart.
+VOLTAGE_EPSILON_FOR_RAW = 1e-1
+
+
+@pytest.fixture
+def ao_single_channel_task(
+    generate_task: nidaqmx.Task, real_x_series_device: nidaqmx.system.Device
+) -> nidaqmx.Task:
+    task = generate_task()
+    chan_index = 0
+    offset = _get_expected_voltage_for_chan(chan_index)
+    chan = task.ao_channels.add_ao_voltage_chan(
+        real_x_series_device.ao_physical_chans[chan_index].name,
+        min_val=0.0,
+        max_val=offset + VOLTAGE_EPSILON,
+    )
+    # forcing the maximum range for binary read scaling to be predictable
+    chan.ao_dac_rng_high = 10
+    chan.ao_dac_rng_low = -10
+
+    # we'll be doing simple on-demand, so start the task now
+    task.start()
+
+    # set the output to a known initial value
+    task.write(0.0)
+
+    return task
+
+
+# todo: this only works for multiplexed
+@pytest.fixture
+def ai_single_channel_loopback_task(
+    generate_task: nidaqmx.Task, real_x_series_device: nidaqmx.system.Device
```

Comment:
The single channel fixture needs a multiplexed device too since it uses `_ao0_vs_aognd`.